<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>TODO for genejam • genejam</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css"><script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet"><script src="pkgdown.js"></script><meta property="og:title" content="TODO for genejam"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-title-body">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">genejam</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.19.900</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="http://github.com/jmw86069/genejam/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>TODO for genejam</h1>
    </div>

<div id="todo-for-genejam" class="section level1">

<div class="section level2">
<h2 id="id_17nov2025">17nov2025<a class="anchor" aria-label="anchor" href="#id_17nov2025"></a></h2>
<ul><li>For efficiency, do not prepare every annotation until necessary. Test with ‘CCL8’, ‘ADCK1’ which only requires “SYMBOL2EG”.</li>
</ul></div>
<div class="section level2">
<h2 id="id_17sep2025">17sep2025<a class="anchor" aria-label="anchor" href="#id_17sep2025"></a></h2>
<ul><li>
<p>DONE. Investigate <code>"Lars"</code> for “org.Mm.eg.db” and <code>ignore.case=TRUE</code>, should return multiple results.</p>
<ul><li>Issue was caused by using <code>match(x, tolower(y))</code> where it would only match the first valid value. Rare cases where different cases are stored, for example: <code>"Lars", "LARS"</code>. There are 1222 such cases of 184641 in org.Mm.egALIAS2EG as of 17sep2025, with 2 or 3 instances of the same lowercase alias, for example: <code>"cal", "Cal", "CAL"</code>. The previous behavior would only return the first aliases associated, which in many cases was already correct - however in other rarer cases there were additional values associated with the alternative-case aliases.</li>
</ul></li>
<li>
<p>Resolve issue when intermediate has values already, and are amended when the gene column search produces results.</p>
<ul><li>Decide whether the existing value should be used without change, or amended with new values.</li>
</ul></li>
<li><p>Consider option to re-collapse expanded columns, e.g. comma-delimited values at input.</p></li>
</ul></div>
<div class="section level2">
<h2 id="id_12aug2025">12aug2025<a class="anchor" aria-label="anchor" href="#id_12aug2025"></a></h2>
<ul><li>Remove jamba Github dependency, point to CRAN instead.</li>
</ul></div>
<div class="section level2">
<h2 id="id_05dec2024">05dec2024<a class="anchor" aria-label="anchor" href="#id_05dec2024"></a></h2>
<ul><li>
<p>Consider shorthand way to specify species:</p>
<ul><li>“Hs”, “Mm” - which mean <code>"org.Hs.eg.db"</code>, <code>"org.Mm.eg.db"</code>
</li>
<li>“All” - use all installed <code>"org.*.eg.db"</code> packages?</li>
</ul></li>
<li>
<p>When input has delimited values, consider returning as-is</p>
<ul><li>Instead make the wide format optional, but default should match the input.</li>
<li>One key difference is <code>empty_rule="original"</code> will include the full input identifier, and not the first split value.</li>
</ul></li>
<li><p>Consider new default: <code>intermediate="ENTREZID"</code></p></li>
<li><p>Consider new default: <code>ignore.case=TRUE</code></p></li>
<li><p>Add examples showing how to use array annotations?</p></li>
<li>
<p>Consider how to include something like biomaRt methods.</p>
<ul><li>I asked Jianying Li for some examples of “goto” queries he uses.</li>
</ul></li>
</ul></div>
<div class="section level2">
<h2 id="id_29jul2024">29jul2024<a class="anchor" aria-label="anchor" href="#id_29jul2024"></a></h2>
<ul><li>
<p>Consider enabling string search methods:</p>
<ul><li>Use case: Find all the genes where SYMBOL begins with <code>"IFN"</code>
</li>
<li>Use case: Find all genes whose GENENAME contains <code>"Receptor"</code>
</li>
<li>It could be essentially wildcard search, except that results would ideally returned on separated rows. No value returning 500 receptors concatenated onto one row. Therefore, this workflow likely requires its own function. It is not: “convert this ID to official ID” in one-to-one manner, which is the goal of <code><a href="reference/freshenGenes.html">freshenGenes()</a></code>.</li>
</ul></li>
<li>
<p>Consider documenting an improved methylation array annotation workflow, or whole transcriptome array workflow.</p>
<ul><li>Start with chip annotation, follow with other Bioconductor annotations.</li>
</ul></li>
</ul></div>
<div class="section level2">
<h2 id="id_11mar2024">11mar2024<a class="anchor" aria-label="anchor" href="#id_11mar2024"></a></h2>
<ul><li>
<p><code><a href="reference/freshenGenes.html">freshenGenes()</a></code> (and variants)</p>
<ul><li><p>Consider option to re-combine multi-column split values back to the source column name. For example when genes are separated into multiple columns when delimited by <code>split</code>, it can create 50+ column for some entries, making the output unwieldy.</p></li>
<li>
<p>Consider changing default argument <code>intermediate="intermediate"</code> to <code>intermediate="ENTREZID"</code>.</p>
<ul><li>This change may break a few older RMarkdown (only for me), however maybe for the best for the default use case.</li>
</ul></li>
<li><p>Bug review: When input contains column values, and <code>intermediate</code> column which already contains partial data, it should only query the column values when the <code>intermediate</code> is empty or blank, per the <code>handle_multiple</code> argument. Currently it appears to override or augment the existing <code>intermediate</code> after also querying the column value.</p></li>
</ul></li>
</ul></div>
<div class="section level2">
<h2 id="id_29feb2024">29feb2024<a class="anchor" aria-label="anchor" href="#id_29feb2024"></a></h2>
<ul><li>
<p>Consider adding <code>biomaRt</code> queries to supplement the fairly limited coverage of EnsEMBL gene/transcript identifiers in the <code>"org.Hs.eg.db"</code> databases in Bioconductor. Many retired entries are not included, or those without 1:1 mapping to EntrezID.</p>
<ul><li>Very unclear how this step could work. Each attribute can be 1:many.</li>
<li>Entries no longer “key” off of ENTREZID.</li>
</ul></li>
</ul></div>
<div class="section level2">
<h2 id="id_23sep2022">23sep2022<a class="anchor" aria-label="anchor" href="#id_23sep2022"></a></h2>
<ul><li>
<p>When input column contains delimiters, it is split into multiple columns with one value each. However, the output <code>data.frame</code> should contain the original colname with original value, in proper order.</p>
<ul><li>consider adding an argument whether to include/hide the split columns</li>
<li>consider adding an argument whether to include/hide the original column</li>
</ul></li>
</ul></div>
<div class="section level2">
<h2 id="id_07sep2022">07sep2022<a class="anchor" aria-label="anchor" href="#id_07sep2022"></a></h2>
<ul><li>
<p>Use case: Search for genes by description</p>
<ul><li>could involve something like pattern matching by grep</li>
<li>For example, find “NF-KB” genes, or “NF-kB”. Does it need to search for the Kappa symbol?</li>
<li>Or search for “histone” or “ubiquitin”.</li>
</ul></li>
<li><p>Consider changing default argument from <code>intermediate="intermediate"</code>, to <code>intermediate="ENTREZID"</code>.</p></li>
<li>
<p>Consider a helper function for microarray annotation.</p>
<ul><li>Annotation of Affymetrix whole exome array probesets.</li>
<li>Annotation of Illumina whole genome tiling array.</li>
<li>Mimic the annotation step for oligo array classes.</li>
</ul></li>
<li>
<p>Consider allowing input <code>SummarizedExperiment</code></p>
<ul><li>use either <code>rownames(se)</code> or <code>rowData_colnames</code> derived from <code>colnames(rowData(se))</code>
</li>
<li>process would extract the relevant <code>data.frame</code> from <code>rowData(se)</code>, then store results back into the <code>se</code> object.</li>
<li>
<code>freshenGenes_se(se, rowData_colnames=NULL)</code> would use <code>rownames(se)</code>
</li>
<li>
<code>freshenGenes_se(se, rowData_colnames=c("AFFYID"))</code> would use <code>rowData(se)[, rowData_colnames]</code>
</li>
</ul></li>
<li><p>Could consider input types such as <code>ExpressionSet</code>, and the oligo array classes that are often hard to annotate easily.</p></li>
</ul></div>
<div class="section level2">
<h2 id="id_30jun2021-complete">30jun2021 (COMPLETE)<a class="anchor" aria-label="anchor" href="#id_30jun2021-complete"></a></h2>
<p>COMPLETE: Optional case-insensitive lookup</p>
<p>Current workflow requires two un-intuitive steps:</p>
<ol style="list-style-type: decimal"><li>Prepare annotation library with <code>get_anno_db(.., ignore.case=TRUE)</code>. This step also requires the full annotation library input, no convenience of providing only <code>"SYMBOL2EG"</code> for example. Also, this step is fairly slow - about 17 seconds to prepare three annotation environments. If not cached, obviously this step adds 17 seconds to every response time.</li>
</ol><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://github.com/jmw86069/genejam" class="external-link">genejam</a></span><span class="op">)</span></span>
<span><span class="va">try_dbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>org.Hs.egSYMBOL2EG<span class="op">=</span><span class="va">org.Hs.egSYMBOL2EG</span>,</span>
<span>   org.Hs.egACCNUM2EG<span class="op">=</span><span class="va">org.Hs.egACCNUM2EG</span>,</span>
<span>   org.Hs.egALIAS2EGorg.Hs.egALIAS2EG<span class="op">=</span><span class="va">org.Hs.egALIAS2EG</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="va">try_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">try_dbs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span><span class="op">{</span></span>
<span>   <span class="fu"><a href="reference/get_anno_db.html">get_anno_db</a></span><span class="op">(</span><span class="va">i</span>, ignore.case<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal"><li>Convert input to lowercase with <code><a href="https://rdrr.io/r/base/chartr.html" class="external-link">tolower()</a></code>, and query using the custom <code>try_list</code> object from step 1.</li>
</ol><p>The solution seems to be the custom function <code><a href="reference/imget.html">genejam::imget()</a></code> with case-insensitive <code><a href="https://rdrr.io/r/base/get.html" class="external-link">mget()</a></code>, with prefix <code>i</code> to indicate case-insensitive.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/freshenGenes3.html">freshenGenes3</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">tolower</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRKB"</span>, <span class="st">"ApoE"</span><span class="op">)</span><span class="op">)</span>, try_list<span class="op">=</span><span class="va">try_list</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="complete-fix-bug-when-colname-intermediate-is-supplied-as-input">COMPLETE: Fix bug when colname “intermediate” is supplied as input<a class="anchor" aria-label="anchor" href="#complete-fix-bug-when-colname-intermediate-is-supplied-as-input"></a></h2>
<div class="section level3">
<h3 id="reproducible-example-0013900-or-less">Reproducible example (0.0.13.900 or less)<a class="anchor" aria-label="anchor" href="#reproducible-example-0013900-or-less"></a></h3>
<pre><code><span><span class="va">idf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Gene<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MINA"</span>, <span class="st">"MINA"</span>, <span class="st">"GABRR3"</span>, <span class="st">"GABRR3"</span><span class="op">)</span>,</span>
<span>   intermediate<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">""</span>, <span class="st">"84864"</span>, <span class="st">""</span>, <span class="st">"200959"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">idf</span></span>
<span><span class="fu">genejam</span><span class="fu">::</span><span class="fu"><a href="reference/freshenGenes2.html">freshenGenes2</a></span><span class="op">(</span><span class="va">idf</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a></h3>
<pre><code>    Gene intermediate intermediate_v1 SYMBOL                                                               GENENAME
1   MINA                        84864
2   MINA        84864           84864  RIOX2                                                  ribosomal oxygenase 2
3 GABRR3                       200959
4 GABRR3       200959          200959 GABRR3 gamma-aminobutyric acid type A receptor rho3 subunit (gene/pseudogene)</code></pre>
<p>It creates a new column <code>"intermediate_v1"</code> then does not use it.</p>
</div>
<div class="section level3">
<h3 id="desired-outcome">Desired outcome<a class="anchor" aria-label="anchor" href="#desired-outcome"></a></h3>
<p>Values should be left as-is in <code>"intermediate"</code> and new values should be used to fill in holes.</p>
<p>Ideally, user can specify the <code>intermediate_colname</code> so its values are recognized, potentially skipping the first step in <code><a href="reference/freshenGenes.html">freshenGenes()</a></code> if there are no remaining colnames in the input data.</p>
<p>An argument <code>intermediate_colname</code> would make the process a bit more transparent.</p>
<p>Also, the colname <code>intermediate_colname</code> should not be split into multiple columns as is done with input data when it contains delimited values.</p>
</div>
<div class="section level3">
<h3 id="new-workflow-version-0014900">New workflow (version 0.0.14.900)<a class="anchor" aria-label="anchor" href="#new-workflow-version-0014900"></a></h3>
<pre><code><span><span class="va">idf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Gene<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MINA"</span>, <span class="st">"MINA"</span>, <span class="st">"GABRR3"</span>, <span class="st">"GABRR3"</span>, <span class="st">"APOE"</span><span class="op">)</span>,</span>
<span>   ENTREZID<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">""</span>, <span class="st">"84864"</span>, <span class="st">""</span>, <span class="st">"200959"</span>, <span class="st">"84864"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">idf</span></span>
<span><span class="fu">genejam</span><span class="fu">::</span><span class="fu"><a href="reference/freshenGenes2.html">freshenGenes2</a></span><span class="op">(</span><span class="va">idf</span>, intermediate<span class="op">=</span><span class="st">"ENTREZID"</span><span class="op">)</span></span>
<span><span class="fu">genejam</span><span class="fu">::</span><span class="fu"><a href="reference/freshenGenes2.html">freshenGenes2</a></span><span class="op">(</span><span class="va">idf</span>, intermediate<span class="op">=</span><span class="st">"ENTREZID"</span>, handle_multiple<span class="op">=</span><span class="st">"first_hit"</span><span class="op">)</span></span>
<span><span class="fu">genejam</span><span class="fu">::</span><span class="fu"><a href="reference/freshenGenes2.html">freshenGenes2</a></span><span class="op">(</span><span class="va">idf</span>, intermediate<span class="op">=</span><span class="st">"ENTREZID"</span>, handle_multiple<span class="op">=</span><span class="st">"first_try"</span><span class="op">)</span></span>
<span><span class="fu">genejam</span><span class="fu">::</span><span class="fu"><a href="reference/freshenGenes2.html">freshenGenes2</a></span><span class="op">(</span><span class="va">idf</span>, intermediate<span class="op">=</span><span class="st">"ENTREZID"</span>, handle_multiple<span class="op">=</span><span class="st">"all"</span><span class="op">)</span></span>
<span><span class="fu">genejam</span><span class="fu">::</span><span class="fu"><a href="reference/freshenGenes2.html">freshenGenes2</a></span><span class="op">(</span><span class="va">idf</span>, intermediate<span class="op">=</span><span class="st">"ENTREZID"</span>, handle_multiple<span class="op">=</span><span class="st">"best_each"</span><span class="op">)</span></span>
<span><span class="fu">genejam</span><span class="fu">::</span><span class="fu"><a href="reference/freshenGenes2.html">freshenGenes2</a></span><span class="op">(</span><span class="va">idf</span>, intermediate<span class="op">=</span><span class="st">"ENTREZID"</span>, include_source<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
<p>Trickier example showing mix of input types:</p>
<pre><code><span><span class="va">idf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Gene<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MINA"</span>, <span class="st">"RIOX2"</span>, <span class="st">"GABRR3"</span>, <span class="st">"GABRR3"</span>, <span class="st">"RIOX2,GABRR3"</span>, <span class="st">""</span><span class="op">)</span>,</span>
<span>   ENTREZID<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">""</span>, <span class="st">"84864"</span>, <span class="st">""</span>, <span class="st">"200959"</span>, <span class="st">""</span>, <span class="st">"84864,200959"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">idf</span></span>
<span><span class="fu">genejam</span><span class="fu">::</span><span class="fu"><a href="reference/freshenGenes2.html">freshenGenes2</a></span><span class="op">(</span><span class="va">idf</span>, intermediate<span class="op">=</span><span class="st">"ENTREZID"</span><span class="op">)</span></span>
<span><span class="fu">genejam</span><span class="fu">::</span><span class="fu"><a href="reference/freshenGenes2.html">freshenGenes2</a></span><span class="op">(</span><span class="va">idf</span>, intermediate<span class="op">=</span><span class="st">"ENTREZID"</span>, include_source<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
<p>Final example showing input with only the <code>intermediate</code> column:</p>
<pre><code><span><span class="va">idf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Gene<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MINA"</span>, <span class="st">"RIOX2"</span>, <span class="st">"GABRR3"</span>, <span class="st">"GABRR3"</span>, <span class="st">"RIOX2,GABRR3"</span>, <span class="st">""</span><span class="op">)</span>,</span>
<span>   ENTREZID<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">""</span>, <span class="st">"84864"</span>, <span class="st">""</span>, <span class="st">"200959"</span>, <span class="st">""</span>, <span class="st">"84864,200959"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">idf</span></span>
<span><span class="fu">genejam</span><span class="fu">::</span><span class="fu"><a href="reference/freshenGenes2.html">freshenGenes2</a></span><span class="op">(</span><span class="va">idf</span><span class="op">[</span>,<span class="st">"ENTREZID"</span>,drop<span class="op">=</span><span class="cn">FALSE</span><span class="op">]</span>, intermediate<span class="op">=</span><span class="st">"ENTREZID"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="allow-entrezid-or-eg-as-optional-input-columns">Allow ENTREZID or EG as optional input column(s)<a class="anchor" aria-label="anchor" href="#allow-entrezid-or-eg-as-optional-input-columns"></a></h2>
<p>I bring this issue to the top on 30mar2021.</p>
<div class="section level3">
<h3 id="problem-space">Problem space<a class="anchor" aria-label="anchor" href="#problem-space"></a></h3>
<p>It is not currently possible to provide ENTREZID values at input, since these values are expected to be in the <code>"intermediate"</code> column after the first step in the process. There is no annotation that takes ENTREZID and returns ENTREZID.</p>
<p>A close by-pass is to supply data with colname <code>"intermediate"</code>, which works but breaks the actual use of <code>"intermediate"</code>.</p>
<ul><li>Allow <code>"ENTREZID"</code> or <code>"EG"</code> as an input column, some mechanism to allow input data to have the intermediate value, without having to query an annotation.</li>
<li>It would essentially mark one or more columns with something like <code>"AsIs"</code> and the values would be delimited and used to populate the <code>"intermediate"</code> column without query.</li>
<li>There could be a special-case <code>ann_lib="ASIS"</code> that would allow values in the <code>"AsIs"</code> columns to pass through, which would give a mechanism to try querying another column first, then using the <code>"ASIS"</code> column as a secondary option.</li>
</ul></div>
</div>
<div class="section level2">
<h2 id="make-searches-case-insensitive">Make searches case-insensitive<a class="anchor" aria-label="anchor" href="#make-searches-case-insensitive"></a></h2>
<ul><li>This change requires something like <code>better_mget()</code> or perhaps more properly <code><a href="reference/imget.html">imget()</a></code>.</li>
<li>The basic mechanism:</li>
</ul><p><code>{r, imget} input &lt;- c("Apoe", "h1-3", "Hist1H1C"); keys &lt;- ls(org.Hs.egSYMBOL2EG) keymatch &lt;- match(tolower(input), tolower(keys)) keysfound &lt;- !is.na(keymatch); values &lt;- as.list(jamba::nameVector(rep(NA, length(input)), input)) valuesfound &lt;- mget(keys[keymatch[keysfound]], org.Hs.egSYMBOL2EG, ifnotfound=NA) values[keysfound] &lt;- valuesfound; values;</code></p>
<p>After some testing, it appears the slow step is retrieving all the keys, which is also performed each time the data is queried, compounding the problem.</p>
<p>Current recommendation is to use <code>get_anno_db(..., ignore.case=TRUE)</code> which returns an environment where all keys are forced to lowercase. In this case, the environment consumes real memory, and takes longer to prepare. However once prepared, the environment can be passed directory to <code><a href="reference/freshenGenes.html">freshenGenes()</a></code> for rapid operation.</p>
<div class="section level3">
<h3 id="suggested-mechanism">Suggested mechanism<a class="anchor" aria-label="anchor" href="#suggested-mechanism"></a></h3>
<ul><li><p><code>asis_colnames=c("ENTREZID","EG")</code> would define colnames <code>c("ENTREZID","EG")</code> as “ASIS” which means they are not to be queried.</p></li>
<li>
<p><code>ann_lib=c("ASIS", "SYMBOL2EG", "ALIAS2EG")</code> would process in order:</p>
<ol style="list-style-type: decimal"><li>take all values from the <code>asis_colnames</code> and delimit them to populate the <code>"intermediate"</code> column.</li>
<li>query <code>"SYMBOL2EG"</code> using all columns except <code>asis_colnames</code>.</li>
<li>query <code>"ALIAS2EG"</code> using all columns except <code>asis_colnames</code>.</li>
</ol></li>
<li>
<p><code>ann_lib=c("SYMBOL2EG", "ASIS", "ALIAS2EG")</code> would process in order:</p>
<ol style="list-style-type: decimal"><li>query <code>"SYMBOL2EG"</code> using all columns except <code>asis_colnames</code>.</li>
<li>Populate <code>"intermediate"</code> with delimited values from the <code>asis_colnames</code> using rules defined by <code>handle_multiple</code>. That is, when <code>handle_multiple="first_try"</code> then only rows with empty values in `“intermediate” column will be populated.</li>
<li>query <code>"ALIAS2EG"</code> using all columns except <code>asis_colnames</code>.</li>
</ol></li>
</ul></div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>



      <footer><div class="copyright">
  <p></p><p>Developed by James M. Ward.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

